react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -2,9 +2,39 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
+import styled from 'styled-components';
 import PurposeButton from './purposeButton';
 import QRPrivacy from '../visitors/qrprivacy';
 import { Activities, Visitors } from '../../api';
+import { Heading } from '../../shared/components/text/base';
+import { FlexContainerRow } from '../../shared/components/layout/base';
+
+const StyledNav = styled.nav`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const StyledSection = styled.section`
+  margin: 3rem 0;
+  display: flex;
+  justify-content: center;
+`;
+
+const HalfFlexContainerRow = FlexContainerRow.extend`
+  width: 50%;
+  padding: 1rem;
+  flex-wrap: wrap;
+`;
+
+const SnakeContainerRow = FlexContainerRow.extend`
+  width: 100%;
+  justify-content: space-between;
+  &:nth-child(2n) {
+  flex-direction: row-reverse;
+  `;
+
+const capitaliseFirstName = name => name.split(' ')[0].replace(/\b\w/g, l => l.toUpperCase());
```

Comment:
First name is first, so this is just capitalising the first character of a string:
```js
const capitalise = str => str.slice(0, 1).toUpperCase().concat(str.slice(1));
```

Reply:
Names are given as a full string eg `"Jessica Jones"`, so I also need to drop the second name. I went for regex over slice as it was marginally less code over `.slice`.